src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/PrisonerApiServiceTest.kt

```diff
@@ -262,6 +263,66 @@ internal class PrisonerApiServiceTest : ServiceTestBase() {
     assertThat(offences[3].offenceDescription).isEqualTo("NMO")
   }
 
+  @Test
+  fun `prison movements which refer to prisonIds referencing inactive prisons are catered for`() {
+    val nomsId = "AB234A"
+
+    val response = mock(PrisonTimelineResponse::class.java)
+
+    given(prisonApiClient.retrievePrisonTimelines(nomsId)).willReturn(
+      Mono.fromCallable {
+        response
+      },
+    )
+
+    given(response.prisonPeriod).willReturn(
+      listOf(
+        PrisonPeriod(
+          bookingId = 123,
+          movementDates = listOf(
+            Movement(releaseFromPrisonId = "MDI"),
+          ),
+        ),
+      ),
+    )
+
+    given(prisonApiClient.retrieveSentencesAndOffences(123)).willReturn(
+      Mono.fromCallable {
+        listOf(
+          Sentence(
+            bookingId = 123,
+            courtDescription = "DEF",
+            sentenceDate = LocalDate.now().minusMonths(3).plusDays(2),
+            sentenceEndDate = LocalDate.now(),
+
+            offences = listOf(
+              SentenceOffence(offenceDescription = "DEF"),
+              SentenceOffence(offenceDescription = "ABC"),
+              SentenceOffence(offenceDescription = "NMO"),
+              SentenceOffence(offenceDescription = "GHI"),
+            ),
+          ),
+        )
+      },
+    )
+
+    given(prisonApiClient.retrieveAgency("MDI"))
+      .willThrow(NotFoundException("Prison api returned agency not found for agency id MDI"))
+
+    val offenderResponse = mock(Offender::class.java)
+
+    given(prisonApiClient.retrieveOffender(any())).willReturn(
+      Mono.fromCallable {
+        offenderResponse
+      },
+    )
+
+    val result = PrisonerApiService(prisonApiClient).retrieveOffences(nomsId)
+
+    val offences = result.get(0).offences
+    assertThat(offences.size).isGreaterThan(0)
```

Comment:
Could you add some assertions here making sure we are still filling in the other fields and not just returning everything as null, please?